<launch>

  <arg name="sensorOffsetX" default="0.0"/>
  <arg name="sensorOffsetY" default="0.0"/>
  <arg name="cameraOffsetZ" default="0.0"/>
  <arg name="twoWayDrive" default="true"/>
  <arg name="maxSpeed" default="1.0"/>
  <arg name="autonomyMode" default="true"/>
  <arg name="autonomySpeed" default="0.5"/>
  <arg name="joyToSpeedDelay" default="1.0"/>
  <arg name="goalCloseDis" default="0.4"/>
  <arg name="adjacentRange" default="3.0"/>
  <arg name="is_real_robot" default="false"/>
  <arg name="laserTopic" default="/utlidar/cloud"/>
  <arg name="laserFixedFrame" default="base_link"/>

  <node pkg="local_lidar_planner" exec="local_lidar_planner_node" name="local_lidar_planner" output="screen">
    <param name="pathFolder" value="$(find-pkg-share local_lidar_planner)/paths" />
    <param name="vehicleLength" value="0.3" />
    <param name="vehicleWidth" value="0.7" />
    <param name="sensorOffsetX" value="$(var sensorOffsetX)" />
    <param name="sensorOffsetY" value="$(var sensorOffsetY)" />
    <param name="cameraOffsetZ" value="$(var cameraOffsetZ)" />
    <param name="twoWayDrive" value="$(var twoWayDrive)" />
    <param name="maxSpeed" value="$(var maxSpeed)" />
    <param name="autonomyMode" value="$(var autonomyMode)" />
    <param name="autonomySpeed" value="$(var autonomySpeed)" />
    <param name="joyToSpeedDelay" value="$(var joyToSpeedDelay)" />
    <param name="goalCloseDis" value="$(var goalCloseDis)" />
    <param name="adjacentRange" value="$(var adjacentRange)" />
    <param name="useTerrainAnalysis" value="false" />
    <param name="laserTopic" value="$(var laserTopic)" />
    <param name="laserFixedFrame" value="$(var laserFixedFrame)" />
  </node>

  <node pkg="local_lidar_planner" exec="path_follower_node" name="path_follower" output="screen">
    <param name="sensorOffsetX" value="$(var sensorOffsetX)" />
    <param name="sensorOffsetY" value="$(var sensorOffsetY)" />
    <param name="twoWayDrive" value="$(var twoWayDrive)" />
    <param name="maxSpeed" value="$(var maxSpeed)" />
    <param name="autonomyMode" value="$(var autonomyMode)" />
    <param name="autonomySpeed" value="$(var autonomySpeed)" />
    <param name="joyToSpeedDelay" value="$(var joyToSpeedDelay)" />
    <param name="goalCloseDis" value="$(var goalCloseDis)" />
    <param name="is_real_robot" value="$(var is_real_robot)" />
  </node>

</launch>
